test/functional/internal/test_utils.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Test suite for ``dynamodb_encryption_sdk.internal.utils``."""
+import copy
+
+import pytest
+
+from dynamodb_encryption_sdk.encrypted import CryptoConfig
+from dynamodb_encryption_sdk.identifiers import CryptoAction
+from dynamodb_encryption_sdk.internal.utils import encrypt_batch_write_item
+from dynamodb_encryption_sdk.material_providers import CryptographicMaterialsProvider
+from dynamodb_encryption_sdk.structures import AttributeActions, EncryptionContext
+from dynamodb_encryption_sdk.transform import ddb_to_dict
+
+from ..functional_test_vector_generators import attribute_test_vectors
+
+
+def get_test_item(standard_dict_format, partition_key, sort_key):
+    attributes = attribute_test_vectors("serialize")
+
+    attributes = {"attr_" + str(pos): attribute[0] for pos, attribute in enumerate(attributes)}
+    attributes["partition-key"] = {"S": partition_key}
+    if sort_key:
+        attributes["sort-key"] = {"S": sort_key}
+
+    if standard_dict_format:
+        attributes = ddb_to_dict(attributes)
+    return attributes
+
+
+def get_test_items(standard_dict_format, table_name="table"):
+    items = [
+        get_test_item(standard_dict_format, partition_key="key-1", sort_key=None),
+        get_test_item(standard_dict_format, partition_key="key-2", sort_key=None),
+        get_test_item(standard_dict_format, partition_key="key-3", sort_key="sort-1"),
+        get_test_item(standard_dict_format, partition_key="key-4", sort_key="sort-1"),
+        get_test_item(standard_dict_format, partition_key="key-4", sort_key="sort-2"),
+    ]
+
+    for pos, item in enumerate(items):
+        item["encrypt-me"] = table_name + str(pos)
+
+    return {table_name: [{"PutRequest": {"Item": item}} for item in items]}
+
+
+def get_dummy_crypto_config(partition_key_name, sort_key_name, encrypted_attribute_name):
+    context = EncryptionContext(partition_key_name=partition_key_name, sort_key_name=sort_key_name)
+    actions = AttributeActions(
+        default_action=CryptoAction.DO_NOTHING,
+        attribute_actions={encrypted_attribute_name: CryptoAction.ENCRYPT_AND_SIGN},
+    )
+    materials = CryptographicMaterialsProvider()
```

Comment:
Just to keep the test surface small, let's use a CMP that will actually work. You can get an ephemeral static CMP from `functional_test_utils.build_static_jce_cmp`[1].

[1] https://github.com/aws/aws-dynamodb-encryption-python/blob/master/test/functional/functional_test_utils.py#L176-L184

Reply:
Sure thing, though I intentionally used an empty CMP here so it didn't confuse the fact that I'm completely faking the encryption in these tests. Is the worry that the base class may not always be so easily constructable for tests - maybe I should just explicitly mock it out instead?